app/models/form_profiles/va10203.rb

```diff
@@ -1,11 +1,88 @@
 # frozen_string_literal: true
 
+module VA10203
+  FORM_ID = '22-10203'
+
+  class FormInstitutionInfo
+    include Virtus.model
+
+    attribute :name, String
+    attribute :city, String
+    attribute :state, String
+  end
+
+  class FormEntitlementInformation
+    include Virtus.model
+
+    attribute :months, Integer
+    attribute :days, Integer
+  end
+end
+
 class FormProfiles::VA10203 < FormProfile
+  attribute :remaining_entitlement, VA10203::FormEntitlementInformation
+  attribute :school_information, VA10203::FormInstitutionInfo
+
+  def prefill(user)
+    authorized = user.authorize :evss, :access?
+
+    if authorized
+      gi_bill_status = get_gi_bill_status(user)
+      @remaining_entitlement = initialize_entitlement_information(gi_bill_status)
+      @school_information = initialize_school_information(gi_bill_status)
+    else
+      @remaining_entitlement = {}
+      @school_information = {}
+    end
+
+    super(user)
+  end
+
   def metadata
     {
       version: 0,
       prefill: true,
       returnUrl: '/applicant/information'
     }
   end
+
+  private
+
+  def get_gi_bill_status(user)
+    service = EVSS::GiBillStatus::Service.new(user)
+    service.get_gi_bill_status
+  rescue => e
+    Rails.logger.error "Failed to retrieve GiBillStatus data: #{e.message}"
+    {}
+  end
+
+  def initialize_entitlement_information(gi_bill_status)
+    return {} if gi_bill_status == {}
+    return {} if gi_bill_status.remaining_entitlement.blank?
```

Comment:
You could make the 'return {}' a one liner by combining the if's with an || or maybe it's more readable the current way?

Reply:
Sounds good.  No preference here!